src/sentry/uptime/consumers/eap_converter.py

```diff
@@ -0,0 +1,172 @@
+    request_sequence: int,
+    item_id: bytes,
+) -> TraceItem:
+    """
+    Convert an individual request to a denormalized UptimeResult TraceItem.
+
+    This creates a TraceItem that includes both check-level metadata, which is duplicated
+    and request-specific data for unified querying.
+
+    In the case of misses, we'll have one row and the request_info will be empty.
+    """
+    attributes: MutableMapping[str, AnyValue] = {}
+
+    attributes["guid"] = _anyvalue(result["guid"])
+    attributes["subscription_id"] = _anyvalue(result["subscription_id"])
+    attributes["check_status"] = _anyvalue(result["status"])
+    if "region" in result:
+        attributes["region"] = _anyvalue(result["region"])
+
+    attributes["scheduled_check_time_us"] = _anyvalue(ms_to_us(result["scheduled_check_time_ms"]))
+    attributes["actual_check_time_us"] = _anyvalue(ms_to_us(result["actual_check_time_ms"]))
+
+    duration_ms = result["duration_ms"]
+    if duration_ms is not None:
+        attributes["check_duration_us"] = _anyvalue(ms_to_us(duration_ms))
+
+    status_reason = result["status_reason"]
+    if status_reason is not None:
+        attributes["status_reason_type"] = _anyvalue(status_reason["type"])
+        attributes["status_reason_description"] = _anyvalue(status_reason["description"])
+
+    if "request_info_list" in result and result["request_info_list"]:
+        first_request = result["request_info_list"][0]
+        attributes["method"] = _anyvalue(first_request["request_type"])
+        if "url" in first_request:
+            # This should always be here once we start passing url, but for backwards compat
+            # we should be cautious here
+            attributes["original_url"] = _anyvalue(first_request["url"])
+
+    attributes["check_id"] = _anyvalue(result["guid"])
+    attributes["request_sequence"] = _anyvalue(request_sequence)
+
+    if request_info is not None:
+        attributes["request_type"] = _anyvalue(request_info["request_type"])
+        http_status_code = request_info["http_status_code"]
+        if http_status_code is not None:
+            attributes["http_status_code"] = _anyvalue(http_status_code)
+
+        if "url" in request_info:
+            attributes["request_url"] = _anyvalue(request_info["url"])
+        if "request_body_size_bytes" in request_info:
+            attributes["request_body_size_bytes"] = _anyvalue(
+                request_info["request_body_size_bytes"]
+            )
+        if "response_body_size_bytes" in request_info:
+            attributes["response_body_size_bytes"] = _anyvalue(
+                request_info["response_body_size_bytes"]
+            )
+        if "request_duration_us" in request_info:
+            attributes["request_duration_us"] = _anyvalue(request_info["request_duration_us"])
+
+        if "durations" in request_info:
+            durations = request_info["durations"]
+            for phase_name in RequestDurations.__annotations__.keys():
```

Comment:
I wonder if there could be a better interface for accessing this on `RequestDurations`. Like `RequestDurations.phases()` or something